### ISSUE TITLE
Bicycle routing: Large detour preferred over routing straight ahead

### ISSUE THREAD

**Issue by scaidermern** [osmandapp/OsmAnd#12911]
In bicycle navigation OsmAnd chooses a really large detour (3.5 km) instead of routing just straight ahead (0.8 km). This can be prevented by setting a single via point:

![photo_2021-10-01_10-11-52](https://user-images.githubusercontent.com/1040784/135587849-2cf4beaa-055e-4bd1-91cb-06e24037c4e8.jpg)

Start: 51.0295467,13.7559128
End: 51.0427633,13.7677574
Route: 
[route.gpx.txt](https://github.com/osmandapp/OsmAnd/files/7265357/route.gpx.txt)

Map data looks fine to me.

Other routing engines don't have this problem:

- [GraphHopper](https://www.openstreetmap.org/directions?engine=graphhopper_bicycle&route=51.0295%2C13.7559%3B51.0429%2C13.7681)
- [OSRM](https://www.openstreetmap.org/directions?engine=fossgis_osrm_bike&route=51.0295%2C13.7559%3B51.0429%2C13.7681)
- [brouter](https://brouter.de/brouter-web/#map=14/51.0375/13.7610/osm-mapnik-german_style&lonlats=13.755913,51.029533;13.767757,51.042784)

**Comment by vshcherb** [comment]
@ivanPyrohivskyi it's not clear why we should make so many changes and does it break something else

**Comment by scaidermern** [comment]
I don't think that the weightings are causing this. Instead it looks like something is causing a high penalty. But that's just a guess.

**Comment by vshcherb** [comment]
Lowering weight for bicycle designated roads doesn't sounds like the right idea, indeed the distance difference around 2 times (3.5 km vs 1.8 km), so we need to check whether it's appropriate to drive through the park and what's appropriate speed
